### PR TITLE
Add deno package.json support

### PR DESCRIPTION
**Pull request by tsuyoshicho**
see https://deno.com/blog/v1.31#packagejson-support
Add deno with package.json support

After deno run(with package.json), generate `node_modules/.deno/` directory and some modules under it.

Add this check.

----

Tried deno with package.json, started server.

Other settings do not test, please review it.

### CODE REVIEW

File: settings/deno.vim [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
     if !empty(lsp#utils#find_nearest_parent_file(lsp#utils#get_buffer_path(), 'deno.jsonc'))
         return []
     endif
 
+    " Deno with package.json, create directory 'node_modules/.deno/' and some module under it.
+    if !empty(lsp#utils#find_nearest_parent_file_directory(lsp#utils#get_buffer_path(), 'node_modules/.deno/'))
+        return []
+    endif
+
+    " Is not non-Deno(=npm) project
```
Comment (heavenshell) [inline settings/deno.vim:15]: Nit picking.
Is not Deno(=npm) project
or
Is non-Deno(=npm) project

Comment (tsuyoshicho) [reply settings/deno.vim:15]: Thanks

But this statement is `empty` check.

Result: npm project mark(`node_modules`) is not exists.

Comment (heavenshell) [reply settings/deno.vim:15]: Thanks. Undestood.
`not non` is bit a confusing to me 😅

Comment (tsuyoshicho) [reply settings/deno.vim:15]: If there is a better comment, I will adopt it, and/or if it is misleading, I will remove it.

**Pull request closed without merging**